src/ui/SearchAlerts/SearchAlerts.ts

```diff
@@ -207,61 +209,87 @@ export class SearchAlerts extends Component {
-          </tr>
-        </tbody>
-      </table>`;
+    const container = $$('div');
+    const table = $$('table', {
+      className: 'coveo-subscriptions-panel-content',
+      cellspacing: 0
+    });
+    container.append(table.el);
+    const tableHead = $$('thead');
+    table.append(tableHead.el);
+
+    const rowHead = $$('tr');
+    tableHead.append(rowHead.el);
+
+    const headerType = $$('th', {
+      className: 'coveo-subscriptions-panel-content-type',
+    }, l('SearchAlerts_Type'));
+    const headerContent = $$('th', null, l('SearchAlerts_Content'));
+    const headerFrequency = $$('th', null, l('SearchAlerts_Frequency'));
+    const headerActions = $$('th', {
+      className: 'coveo-subscriptions-panel-content-actions'
+    }, l('SearchAlerts_Actions'));
+
+    rowHead.append(headerType.el);
+    rowHead.append(headerContent.el);
+    rowHead.append(headerFrequency.el);
+    rowHead.append(headerActions.el);
+
+    const tableBodySpacer = $$('tbody', {
+      className: 'coveo-subscriptions-panel-spacer'
+    }, $$('tr', null, $$('td', {
+      colsspan: 3
+    })));
+
+    table.append(tableBodySpacer.el);
+
+    const tableBodySubscriptions = $$('tbody', {
+      className: 'coveo-subscriptions-panel-subscriptions'
+    }, $$('tr', {
+      className: 'coveo-subscriptions-panel-no-subscriptions'
+    }, $$('td', {
+      colspan: 3
+    }, l('SearchAlerts_PanelNoSearchAlerts'))));
+
+    table.append(tableBodySubscriptions.el);
+    let sizeModForModalBox = 'big';
 
     return this.queryController.getEndpoint().listSubscriptions().then((subscriptions: ISubscription[]) => {
       _.each(subscriptions, (subscription) => {
         this.addSearchAlert(subscription, container);
       });
-    })
-      .catch(() => {
-        container.el.innerHTML = '<div class=\'coveo-subscriptions-panel-fail\'>' + l('SearchAlerts_Fail') + '</div>';
-      })
-      .finally(() => {
-        this.modal = ModalBox.open(container.el, {
-          titleClose: false,
-          overlayClose: true,
-          title: title.el.outerHTML,
-          className: 'coveo-subscriptions-panel',
-          sizeMod: 'small'
-        });
-
+    }).catch(() => {
+      sizeModForModalBox = 'small';
+      container.empty();
+      container.append(this.getFailureMessage().el);
+    }).finally(() => {
+      this.modal = this.ModalBox.open(container.el, {
+        title: title.el.outerHTML,
+        className: 'coveo-subscriptions-panel',
+        sizeMod: sizeModForModalBox
       });
+    });
+  }
+
+  private getFailureMessage(): Dom {
+    return $$('div', {
+      className: 'coveo-subscriptions-panel-fail'
+    }, l('SearchAlerts_Fail'));
   }
 
   private handleSearchAlertsFail() {
-    this.close();
     if (this.modal != null) {
-      this.modal.content.innerHTML = '<div class=\'coveo-subscriptions-panel-fail\'>' + l('SearchAlerts_Fail') + '</div>';
+      const modalBody = $$(this.modal.wrapper).find('.coveo-modal-body');
```

Comment:
Do you really need to flush the whole modal on an error? :/

I think it could either display a small modal or append a section in the current modal.